app/src/main/java/com/rootstrap/android/util/LoadingDialogUtil.kt

```diff
@@ -0,0 +1,41 @@
+package com.rootstrap.android.util
+
+import android.app.AlertDialog
+import android.content.Context
+import com.rootstrap.android.R
+import com.rootstrap.android.ui.custom.LoadingDialog
+
+object LoadingDialogUtil {
+
+    private var loadingDialog: LoadingDialog? = null
+
+    fun showProgress(context: Context) {
+        if (loadingDialog == null) {
+            loadingDialog = LoadingDialog(context, null)
+        }
+
+        loadingDialog?.show()
+    }
+
+    fun hideProgress() {
+        loadingDialog?.run { dismiss() }
+    }
+
+    fun showError(message: String?, context: Context) {
+        val builder = AlertDialog.Builder(context)
+        builder.setTitle(context.getString(R.string.error))
+
+        when (message) {
+            "" -> builder.setMessage(context.getString(R.string.generic_error))
+            null -> builder.setMessage(context.getString(R.string.generic_error))
+            else -> builder.setMessage(message)
+        }
```

Comment:
try with `message.isNullOrEmpty()`
```suggestion
        message = if(message.isNullOrEmpty()) 
            context.getString(R.string.generic_error)
        else
            message    
       
       builder.setMessage(message)

```

Reply:
done!